src/components/text.js

```diff
@@ -0,0 +1,333 @@
+var registerComponent = require('../core/component').registerComponent;
+var THREE = require('../lib/three');
+
+var createTextGeometry = require('three-bmfont-text');
+var loadBMFont = require('load-bmfont');
+var path = require('path');
+var assign = require('object-assign');
+var createSDF = require('three-bmfont-text/shaders/sdf');
+var createMSDF = require('three-bmfont-text/shaders/msdf');
+var createBasic = require('three-bmfont-text/shaders/basic');
+
+var coreShader = require('../core/shader');
+var shaders = coreShader.shaders;
+
+var alignments = ['left', 'right', 'center'];
+var anchors = ['left', 'right', 'center', 'align'];
+var baselines = ['top', 'center', 'bottom'];
+
+var DEFAULT_WIDTH = 1; // 1 matches other AFRAME default widths... 5 matches prior bmfont examples etc.
+
+// @bryik set anisotropy to 16 because I think it improves the look
+// of large amounts of text particularly when viewed from an angle.
+var MAX_ANISOTROPY = 16;
+
+var FONT_BASE_URL = 'https://cdn.aframe.io/fonts/';
+var fontMap = {
+  'default': FONT_BASE_URL + 'DejaVu-sdf.fnt',
+
+  'aileronsemibold': FONT_BASE_URL + 'Aileron-Semibold.fnt',
+  'dejavu': FONT_BASE_URL + 'DejaVu-sdf.fnt',
+  'exo2bold': FONT_BASE_URL + 'Exo2Bold.fnt',
+  'exo2semibold': FONT_BASE_URL + 'Exo2SemiBold.fnt',
+  'kelsonsans': FONT_BASE_URL + 'KelsonSans.fnt',
+  'monoid': FONT_BASE_URL + 'Monoid.fnt',
+  'mozillavr': FONT_BASE_URL + 'mozillavr.fnt',
+  'sourcecodepro': FONT_BASE_URL + 'SourceCodePro.fnt'
+};
+
+var loadedFontPromises = {};
+var loadedTexturePromises = {};
+
+module.exports.Component = registerComponent('text', {
+  multiple: true,
+
+  schema: {
+    align: {type: 'string', default: 'left', oneOf: alignments},
+    alphaTest: {default: 0.5},
+    // center default to match primitives like plane; if 'align', null or undefined, same as align
+    anchor: {default: 'center', oneOf: anchors},
+    baseline: {default: 'center', oneOf: baselines},
+    color: {type: 'color', default: '#000'},
+    font: {type: 'string', default: 'default'},
+    // default to fnt but with .fnt replaced by .png
+    fontImage: {type: 'string'},
+    // no default, will be populated at layout
+    height: {type: 'number'},
+    letterSpacing: {type: 'number', default: 0},
+    // default to font's lineHeight value
+    lineHeight: {type: 'number'},
+    opacity: {type: 'number', default: '1.0'},
+    shader: {default: 'modified-sdf', oneOf: ['modified-sdf', 'sdf', 'basic', 'msdf']},
+    side: {default: 'front', oneOf: ['front', 'back', 'double']},
+    tabSize: {default: 4},
+    transparent: {default: true},
+    value: {type: 'string'},
+    whiteSpace: {default: 'normal', oneOf: ['normal', 'pre', 'nowrap']},
+    // default to geometry width, or if not present then DEFAULT_WIDTH
+    width: {type: 'number'},
+    // units are 0.6035 * font size e.g. about one default font character (monospace DejaVu size 32)
```

Comment:
Following the discussion about the arbitrary `0.6035` number. I would remove it from the code as it's just an specific case hand-made computed value used to match the DejaVu size, so if you change the font it doesn't make any sense to keep using it and the values for width will be also arbitrary. 
So better to skip using it and make the code less cryptic.